### ISSUE TITLE
Update the ARAX container to ubuntu 20.04

### ISSUE THREAD

**Issue by finnagin** [RTXteam/RTX#1686]
Pretty self explanatory. Hopefully doing so would make our system more  robust to issues like the lets encrypt cert expiration that happened today. Also could be a good excuse to clear out some of the  cruft in our current container. We may want to decide what we are going to do in issue #1685 as a part of this as well.

**Comment by edeutsch** [comment]
Yes, I like the idea of this. Should we also migrate the host OS to Ubuntu 20 while we're at it?
- Set up a fresh Ubuntu 20 instance and do all updates, install Docker, etc.
- Set up our system based on what we learned from deploying to ITRB all fresh in a container running U20
- Copy only the cruft that we want to preserve from the old instance so that we end up pretty clean
- Then make that the main server and retire the old
- We'll need to make some Python and Python package version decisions along the way

What is ITRB instance running on? Is that all Ubuntu 20? or?

**Comment by finnagin** [comment]
Hmmm I'm not sure. Since we are on a NCATS server wouldn't he have to go through them to reinstall the OS or to stand up a new server if we take that route? 

I have no idea what ITRB is running. Their setup is kept pretty opaque from my end.

**Comment by edeutsch** [comment]
> Hmmm I'm not sure. Since we are on a NCATS server wouldn't he have to go through them to reinstall the OS or to stand up a new server if we take that route?

True. Maybe worth the effort?

> I have no idea what ITRB is running. Their setup is kept pretty opaque from my end.

Remarkable!

**Comment by saramsey** [comment]
Whatever we do, I think we should have the host OS on `arax-backup.rtx.ai` and `arax.ncats.io` be the exact same OS version. Also, it may make sense for us to test out Ubuntu 20.04 on a throwaway AWS instance to make sure we are happy with how Docker, nginx, OpenSSL, and OpenSSH run in it.

**Comment by edeutsch** [comment]
Although I may have been responsible for the urge to modernize to Ubuntu 20, since learning that ITRB is all on Ubuntu 18. I am now inclined to consolidate to Ubuntu 18 so that our setup is most similar to theirs. So I'm thinking that migrating to a fresh Ubuntu 18 container makes sense, but maybe we should forget about Ubuntu 20 for now.

**Comment by dkoslicki** [comment]
@finnagin Is this issue still relevant?

**Comment by finnagin** [comment]
I think so. We are currently using Ubuntu 20.04 for the ITRB and arax-backup deployments, but are still using Ubuntu 16.04 for the production deployment. We have not changed the production deployment yet as we did not want to disrupt the container before the demo. 

One last thing we will still need to do before we can switch is take stock of the endpoints in the container on arax.ncats.io and alter the dockerfile to mimic those. Otherwise I think we could make the switch whenever we want though we should definitely save a copy of an image of the current container and a copy of everything in /translator/data on arax.ncats.io before replacing it with a new container.

When we do decide to make the switch I would propose that we pick a date and time that everyone could be online.

One last complication is if we also want to update the host os. Then I think it might make sense to do that at the same time but that also requires coordinating with NCATS as they own the EC2 instance. 

I think maybe we should discuss at the first AHM after the relay next week.

**Comment by dkoslicki** [comment]
To discuss on AHM if we want to update prod from 16.04 to 20.04

**Comment by dkoslicki** [comment]
From 6/22 AHM: lower priority given other impending deadlines

**Comment by saramsey** [comment]
This will happen automatically when we transition to `arax2.rtx.ai`, which is #2129.
